packages/api/utils/response.ts

```diff
@@ -137,3 +137,84 @@ export class ErrorResponse extends Response {
   }
 }
+
+/**
+ * To be used when a database error occurs.  Hides error/implementation details.
+ */
+export const DATABASE_ERROR = new ErrorResponse(
+  { code: ErrorCodes.OTHER, message: "Database error" },
+  ErrorStatusCode.INTERNAL_SERVER_ERROR
+);
+
+/**
+ * To be used when a request body is required but was not provided
+ */
+export const REQUEST_BODY_EMPTY = new ErrorResponse(
+  { code: ErrorCodes.INVALID_INPUT, message: "Request body must not be empty" },
+  ErrorStatusCode.BAD_REQUEST
+);
+
+/**
+ * To be used when a request body exists but is invalid
+ * Could be invalid because:
+ *  - request body is not of the expected Content-Type (for example, application/json, or application/pdf)
+ *  - is not valid JSON / PDF
+ *  - when JSON, is not of the expected type (for example, doesn't look like PortfolioStep/FundingStep/ApplicationStep)
+ */
+export const REQUEST_BODY_INVALID = new ErrorResponse(
+  { code: ErrorCodes.INVALID_INPUT, message: "A valid request body must be provided" },
+  ErrorStatusCode.BAD_REQUEST
+);
+
+/**
+ * To be used when a request parameter exists but is invalid
+ * Could be invalid because:
+ *  - has the wrong form (for example, uuid or date expected but won't parse)
+ *  - is out of range (for example, integer not in the accepted range)
+ */
+export const QUERY_PARAM_INVALID = new ErrorResponse(
+  { code: ErrorCodes.INVALID_INPUT, message: "Invalid request parameter" },
+  ErrorStatusCode.BAD_REQUEST
+);
+
+/**
+ * To be used when an expected path variable exists but is invalid
+ * Could be invalid because:
+ *  - has the wrong form (for example, uuid expected but won't parse)
+ */
+export const PATH_VARIABLE_INVALID = new ErrorResponse(
+  { code: ErrorCodes.INVALID_INPUT, message: "Invalid path variable" },
+  ErrorStatusCode.BAD_REQUEST
+);
+
+/**
+ * To be used when the specified Portfolio Draft is not found
+ */
+export const NO_SUCH_PORTFOLIO_DRAFT = new ErrorResponse(
+  { code: ErrorCodes.INVALID_INPUT, message: "Portfolio Draft with the given ID does not exist" },
+  ErrorStatusCode.NOT_FOUND
+);
+
+/**
+ * To be used when a Portfolio Step is not found for a specified Portfolio Draft
+ */
+export const NO_SUCH_PORTFOLIO_STEP = new ErrorResponse(
+  { code: ErrorCodes.INVALID_INPUT, message: "Portfolio Step not found for this Portfolio Draft" },
+  ErrorStatusCode.NOT_FOUND
+);
+
+/**
+ * To be used when a Funding Step is not found for a specified Portfolio Draft
+ */
+export const NO_SUCH_FUNDING_STEP = new ErrorResponse(
+  { code: ErrorCodes.INVALID_INPUT, message: "Funding Step not found for this Portfolio Draft" },
+  ErrorStatusCode.NOT_FOUND
+);
+
+/**
+ * To be used when a Application Step is not found for a specified Portfolio Draft
+ */
+export const NO_SUCH_APPLICATION_STEP = new ErrorResponse(
+  { code: ErrorCodes.INVALID_INPUT, message: "Application Step not found for this Portfolio Draft" },
+  ErrorStatusCode.NOT_FOUND
+);
```

Comment:
I am tempted to recommend moving these outside of this file. `utils/response.ts` largely defines more general structures to be used, mostly free of specific application text or data. Maybe these can go in a `utils/errors.ts`? I could be talked out of that recommendation if we found a way to stop exporting `ErrorResponse` to force usage of these constants but I am not sure that we're ready for that.

In any case, I think it'd be good to try to keep this file more "abstract" and free from the specific application logic. I can envision a scenario where we might want to "share" this `reponse.ts` file (and perhaps `models/errors.ts`) across multiple applications.

Reply:
My first inclination was to put it in `errors.ts`, but I found most of the supporting code in `response.ts`.  I think the recommendation makes sense.

Reply:
New code relocated to `utils/errors.ts` in 162053ae3e3909bde7f2640a6b70125d8a47a8e5